### DISCUSSION TITLE
Is there no time delay in FAST wind generator.

### DISCUSSION THREAD

**Discussion by Sunwannasleep** [OpenFAST/openfast#1351] [General]
I use the output parameters of FAST for system identification. The input is rotor torque 'RotTorq' and the output is generator electromagnetic power 'GenPwr'. Theoretically, there must be a time delay from the mechanical torque on the shaft to the output electromagnetic power, which is inevitable in practice. But comparing the curves of the two, they change synchronously, as shown in Figure 1.
![QQ图片20221126212055](https://user-images.githubusercontent.com/111493977/204091243-a08e01e8-d495-4294-a6a8-911b6d753eb4.png)

Use the system identification toolbox of MATLAB to identify the above process, as shown in Figure 2. The result shows that there is no time delay, which is not in line with the actual situation.
![QQ图片20221126212346](https://user-images.githubusercontent.com/111493977/204091249-432ebc32-7a8d-499e-876e-1dd2f84839d7.png)

Whether the time delay in the actual wind power generation process can be simulated in FAST, and if not, how can I add the delay？

Best regards.

**Comment by jjonkman** [comment]
Dear @Sunwannasleep,

Which variable speed controller or generator model have you enabled in OpenFAST?  The simple built-in models compute the electrical generator power very simplistically, e.g. torque * speed * efficiency.  The electrical generator power output is not really needed by OpenFAST (it is the mechanical dynamics that is important to capture properly within OpenFAST), but is computed mostly for reference purposes.  If you want to model a more realistic mechanical-to-electrical conversion, you can implement such a model yourself through the user-defined functionality available (user-defined subroutine, controller dynamic library, Simulink interface, etc.).

FYI: The following NREL report may be of interest to you: https://www.nrel.gov/docs/fy14osti/59195.pdf.

Best regards,

**Comment by Sunwannasleep** [comment]
Dear @jjonkman 
This work is carried out under the stable wind of 8m/s, and the generator model part  implemented  through the simulink interface, using the example model _Test01_SIG_ OC4.mdl_  in FAST.
I will try to build a specific generator power control simulation circuit through MATLAB/simulink, Thanks a lot.
Best regards,
